examples/cryptodata-plugin/main.go

```diff
@@ -0,0 +1,143 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"log"
+	"github.com/ligato/cn-infra/agent"
+	"github.com/ligato/cn-infra/db/keyval/etcd"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/servicelabel"
+	"github.com/ligato/cn-infra/db/cryptodata"
+	"github.com/ligato/cn-infra/config"
+	"io/ioutil"
+	"encoding/pem"
+	"crypto/x509"
+	"crypto/rsa"
+	"github.com/pkg/errors"
+	"encoding/base64"
+	"fmt"
+)
+
+// PluginName represents name of plugin.
+const PluginName = "example"
+
+func main() {
+	// Start Agent with ExamplePlugin using ETCDPlugin CryptoDataPlugin, logger and service label.
+	p := &ExamplePlugin{
+		Deps: Deps{
+			Log:          logging.ForPlugin(PluginName),
+			ServiceLabel: &servicelabel.DefaultPlugin,
+			CryptoData:   &cryptodata.DefaultPlugin,
+		},
+	}
+
+	if err := agent.NewAgent(agent.AllPlugins(p)).Run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Deps lists dependencies of ExamplePlugin.
+type Deps struct {
+	Log          logging.PluginLogger
+	ServiceLabel servicelabel.ReaderAPI
+	CryptoData   cryptodata.Client
+}
+
+// ExamplePlugin demonstrates the usage of datasync API.
+type ExamplePlugin struct {
+	Deps
+}
+
+// String returns plugin name
+func (plugin *ExamplePlugin) String() string {
+	return PluginName
+}
+
+// Init starts the consumer.
+func (plugin *ExamplePlugin) Init() error {
+	// Read public key
+	bytes, err := ioutil.ReadFile("../cryptodata-lib/key-pub.pem")
+	if err != nil {
+		return err
+	}
+	block, _ := pem.Decode(bytes)
+	if block == nil {
+		return errors.New("failed to decode PEM for key key-pub.pem")
+	}
+	pubInterface, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return err
+	}
+	publicKey, ok := pubInterface.(*rsa.PublicKey)
+	if !ok {
+		return errors.New("failed to convert public key to rsa.PublicKey")
+	}
+
+	// Create ETCD connection
+	etcdFileConfig := &etcd.Config{}
+	err = config.ParseConfigFromYamlFile("etcd.conf", etcdFileConfig)
+	if err != nil {
+		return err
+	}
+
+	etcdConfig, err := etcd.ConfigToClient(etcdFileConfig)
+	if err != nil {
+		return err
+	}
+
+	db, err := etcd.NewEtcdConnectionWithBytes(*etcdConfig, plugin.Log)
+	if err != nil {
+		return err
+	}
+
+	// Wrap ETCD connection with crypto layer
+	dbWrapped := plugin.CryptoData.Wrap(db, cryptodata.NewDecrypterJSON())
+
+	// Prepare data
+	value := "hello-world"
+	encryptedValue, err := plugin.CryptoData.EncryptData([]byte(value), publicKey)
+	if err != nil {
+		return err
+	}
+	encryptedBase64Value := base64.URLEncoding.EncodeToString(encryptedValue)
+	encryptedJSON := fmt.Sprintf(`{"encrypted":true,"value":{"payload":"$crypto$%v"}}`, encryptedBase64Value)
+	plugin.Log.Infof("Putting value %v", encryptedJSON)
+
+	// Put and get
+	key := plugin.etcdKey("value")
+	err = dbWrapped.Put(key, []byte(encryptedJSON))
```

Comment:
Use just raw `db` for putting the encrypted data to not confuse readers, since `dbWrapped` doesn't actually do anything extra with the data.

Then you can create `dbWrapped` just before the actual `GetValue`.